### ISSUE TITLE
You might be able to avoid some usages of the `sh` module

### ISSUE THREAD

**Issue by subsetpark** [hdrz/bagatto-story#1]
Taking a look at this entry in the site object:

```janet
           # Copy all static content to generated site
           # (sh/$< cp -r "static" "site/")
```

If all you need to do is copy an existing file into `site`, then I believe you can use the built-in DSL in `site`. Here's an example from my blog:

```janet
(defn html? [_site item] (string/has-suffix? ".html" (item :path)))
# ...
(defn uglify-css [site {:contents contents}]
  (sh/$< echo ,(string contents) | minify --type css))
# ...
(def data {:blog {:src "subsetpark.jdn" :attrs bagatto/parse-jdn}
           # ...
           :images {:src (bagatto/* "images/*") :attrs bagatto/parse-base}
           :current {:src (bagatto/* "images/current/*") :attrs bagatto/parse-base}
           :assets {:src (bagatto/slurp-* "assets/*")}})

(def site {:index {:dest "index.html"
                   :out (renderer "/temple/index" {:root ""})}
           # ...
           :images {:each :images :dest (bagatto/path-copier "images")}
           :current {:each :current :dest (bagatto/path-copier "images/current")}
           :static {:each :assets
                    :dest (bagatto/path-copier "static")
                    :filter html?}
           :assets {:each :assets
                    :dest (bagatto/path-copier "assets")
                    :out uglify-css
                    :filter |(not (html? $0 $1))}
            # ...
})
```

Here we see a few different patterns:

1. `images` and `current` are produced with `bagatto/*`, which simply produces an array listing every filename that matches the glob;
2. `assets` is produced with `bagatto/slurp-*`, which actually reads the files matching the glob into strings.
3. `images` and `current` are both copied into the source directory. 
     a. `:each :images` iterates over every item in that data entry;
     b. `:dest (bagatto/path-copier "images")` takes the filename of each item and constructs a path attaching it to `images/`.
4. `:static` is produced by iterating over `assets`, uses the same `:dest` strategy as the file copiers---but adds a `filter` predicate so only certain files get copied over.
5. `:assets` does the same iteration, but includes an `:out` key as well. The item is passed to `uglify-css` and the contents are splatted into an invocation of `minify`. Though of course, because the contents are part of the item, you could pass them to pure Janet if you wanted to avoid shelling out at all.

**Comment by hdrz** [comment]
Yes I started by using `bagatto/*` and `bagatto/slurp-*`, but then changed to the simplified form of just copying the whole `static/` folder.  (is using external linux commands considered bad design? seems ok to me..)

The reason is that eventually I want to copy this whole folder, so why do it subfolder at a time? If I could use something like `bagatto/*` but recursively, it would be ideal.

Now the `assets/` folder handling is more interesting, since there is transformation involved, be it `uglify`/`minify`/`gzip` or something else. The static content starts as assets, then gets transformed into it's final state in the `static/` folder.
In hugo workflow, this transformation happens every build cycle, but my feeling is that this content doesn't change much, hence the relevant lines are commented in `index.janet` and can be enabled anytime they are needed.

Ideally, also the pics for the gallery would be placed in `assets/`, then resized by a call to imagemagik or something similar and placed in `static/` as well.

I see your point about using the built-in commands of the tool at-hand, it would be ideal. Just thoughts about which direction to take..
